src/main/scala/cromwell/engine/backend/jes/Run.scala

```diff
@@ -25,7 +25,10 @@ object Run  {
 
   def apply(pipeline: Pipeline): Run = {
     val rpr = new RunPipelineRequest().setPipelineId(pipeline.id).setProjectId(pipeline.projectId).setServiceAccount(JesServiceAccount)
-    val tag = s"JES Run [UUID(${pipeline.workflow.shortId}):${pipeline.key.scope.name}]"
+    def tag: String = {
+      val shard = pipeline.key.index.map(x => s":$x").getOrElse("")
+      s"JES Run [UUID(${pipeline.workflow.shortId}):${pipeline.key.scope.name}$shard]"
+    }
```

Comment:
Do you want to be second reviewer?  If I'm understanding you correctly, you agree with Miguel that `IndexEnhancedIndex` should have a `toString` method that returns a `":<int>"` or `""`?